drivers/windows/file_access_windows.cpp

```diff
@@ -356,15 +356,40 @@ uint64_t FileAccessWindows::_get_modified_time(const String &p_file) {
 		file = file.substr(0, file.length() - 1);
 	}
 
-	struct _stat st;
-	int rv = _wstat((LPCWSTR)(file.utf16().get_data()), &st);
+	HANDLE handle = CreateFileW((LPCWSTR)(file.utf16().get_data()), GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, 0, NULL);
 
-	if (rv == 0) {
-		return st.st_mtime;
-	} else {
-		print_verbose("Failed to get modified time for: " + p_file + "");
-		return 0;
+	if (handle != INVALID_HANDLE_VALUE) {
+		FILETIME ft_create, ft_write;
+
+		bool status = GetFileTime(handle, &ft_create, nullptr, &ft_write);
```

Comment:
[According to the docs](https://learn.microsoft.com/en-us/windows/win32/sysinfo/file-times), `GetFileTime` can have issues with DST as well (at least with FAT):

> The FAT file system records times on disk in local time. [GetFileTime](https://learn.microsoft.com/en-us/windows/desktop/api/FileAPI/nf-fileapi-getfiletime) retrieves cached UTC times from the FAT file system. When it becomes daylight saving time, the time retrieved by `GetFileTime` is off an hour, because the cache is not updated. When you restart the computer, the cached time that `GetFileTime` retrieves is correct.

But for `_wstat` it [seems to always invalid on FAT](https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2013/14h5k7ff(v=vs.120)), so I guess it's better

> st_atime
> Time of last access of file. Valid on NTFS but not on FAT formatted disk drives.

> st_ctime
> Time of creation of file. Valid on NTFS but not on FAT formatted disk drives.

Reply:
I'm not sure what you are suggesting?